buku

```diff
@@ -1064,58 +1250,87 @@ class BukuDb:
 
                 cond.acquire()
                 if resultset:
-                    row = resultset.pop()
+                    id, url, tags, flags = resultset.pop()
                 else:
                     cond.release()
                     break
                 cond.release()
 
-                title, desc, tags, mime, bad = network_handler(row[1], row[2] & 1)
+                result = fetch_data(custom_url or url, http_head=(flags & FLAG_IMMUTABLE) > 0)
                 count += 1
 
                 cond.acquire()
 
-                if bad:
-                    print(bad_url_str % row[0])
+                if result.bad:
+                    print(bad_url_str % id)
+                    if custom_tags:
+                        self.cur.execute('UPDATE bookmarks SET tags = ? WHERE id = ?', (custom_tags, id))
```

Comment:
Due to semantics of `custom_tags`, it needs to be applied _always_ (regardless of fetch operation success)